fairlearn/reductions/_moments/conditional_selection_rate.py

```diff
@@ -36,12 +37,23 @@ class ConditionalSelectionRate(ClassificationMoment):
     where error(A = a) = total_error
     """
 
-    def __init__(self, ratio=1.0):
+    def __init__(self, difference_bound=None, ratio_bound=None, ratio_bound_slack=0.0):
         """Initialize with the ratio value."""
         super(ConditionalSelectionRate, self).__init__()
-        if not (0 < ratio <= 1):
-            raise ValueError(_MESSAGE_RATIO_NOT_IN_RANGE)
-        self.ratio = ratio
+        if (difference_bound is None) and (ratio_bound is None):
+            self.eps = _DEFAULT_DIFFERENCE_BOUND
+            # TODO: figure out if this initialization of ratio is correct
```

Comment:
To clarify @romanlutz, we can remove this branching statement if we set difference bounr to _DEFAULT_DIFFERENCE_BOUND?

Reply:
With the current logic, If I were to set difference_bound to _DEFAULT_DIFFERENCE_BOUND then I would get a ValueError if we pass in a value for ratio_bound because both ratio_bound and difference_bound would not be None. 

```python
        if difference_bound and ratio_bound:
            raise ValueError("Only one of difference_bound and ratio_bound can be used.")
```

Should I still initialize difference_bound like you suggested and change this logic somehow?

Reply:
The current logic is:
```python
        if (difference_bound is None) and (ratio_bound is None):
            self.eps = _DEFAULT_DIFFERENCE_BOUND
            self.ratio = 1.0
        if difference_bound is not None and ratio_bound is not None:
            raise ValueError(_MESSAGE_INVALID_BOUNDS)
```
but I can change it to
``` python
        if difference_bound is not None and ratio_bound is not None:
            raise ValueError(_MESSAGE_INVALID_BOUNDS)
        elif (difference_bound is None) and (ratio_bound is None):
            difference_bound = _DEFAULT_DIFFERENCE_BOUND
```
Do you think that makes sense?

Reply:
Oh oops yes I completely missed that case! You're right, of course!

Reply:
Yes, sorry I got confused with the ratio_bound_slack for a second. Updated the bound initializations as suggested